core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -333,6 +347,122 @@ public long nextFetchOffset() {
         }
     }
 
+    void updateOffsetsOnLsoMovement() {
```

Comment:
Please mark it visible for testing if the scope is narrowed because of tests.

Reply:
This function is being accessed in the share partition manager. Hence, an object of Share Partition should be able to access it which won't be possible with "private" access